example/src/App.tsx

```diff
@@ -0,0 +1,115 @@
+import { Button, StyleSheet, Text, View } from 'react-native'
+import Spokestack, {
+  SpokestackPlayEvent,
+  SpokestackRecognizeEvent
+} from 'react-native-spokestack'
+
+import React from 'react'
+import checkPermission from './checkPermission'
+
+export default function App() {
+  const [listening, setListening] = React.useState(false)
+  const [playing, setPlaying] = React.useState(false)
+  const [transcript, setTranscript] = React.useState('')
+  const [partial, setPartial] = React.useState('')
+  const [error, setError] = React.useState('')
+
+  const onActivate = () => setListening(true)
+  const onDeactivate = () => setListening(false)
+  const onPlay = ({ playing }: SpokestackPlayEvent) => setPlaying(playing)
+  const onRecognize = ({ transcript }: SpokestackRecognizeEvent) =>
+    setTranscript(transcript)
+  const onRecognizePartial = ({ transcript }: SpokestackRecognizeEvent) =>
+    setPartial(transcript)
+
+  async function init() {
+    const clientId = process.env.SPOKESTACK_CLIENT_ID
+    const clientSecret = process.env.SPOKESTACK_CLIENT_SECRET
+
+    if (!clientId) {
+      throw new Error('SPOKESTACK_CLIENT_ID should be set in the environment.')
+    }
+    if (!clientSecret) {
+      throw new Error(
+        'SPOKESTACK_CLIENT_SECRET should be set in the environment.'
+      )
+    }
+    if (!(await checkPermission())) {
+      setError('Microphone permission is required.')
+      return
+    }
+    Spokestack.initialize(clientId, clientSecret)
+      .then(Spokestack.start)
+      .catch((error) => {
+        setError(error.message)
+      })
+  }
+
+  React.useEffect(() => {
+    Spokestack.addEventListener('recognize', onRecognize)
+    Spokestack.addEventListener('partial_recognize', onRecognizePartial)
+    Spokestack.addEventListener('activate', onActivate)
+    Spokestack.addEventListener('deactivate', onDeactivate)
+    Spokestack.addEventListener('play', onPlay)
+
+    init()
+
+    return () => {
+      Spokestack.removeAllListeners()
+    }
+  }, [])
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.buttons}>
+        <Button
+          title={listening ? 'Listening...' : 'Listen'}
+          onPress={async () => {
+            if (listening) {
+              Spokestack.deactivate()
+            } else {
+              Spokestack.activate()
+            }
+          }}
+          color="#2f5bea"
+        />
+        <Button
+          title={playing ? 'Playing...' : `Play "Hello"`}
+          onPress={() => {
+            Spokestack.speak('Hello World')
+          }}
+        />
+      </View>
+      {!!error && <Text style={styles.error}>{error}</Text>}
+      <View style={styles.results}>
+        <Text>Partial: "{partial}"</Text>
+        <Text>Completed: "{transcript}"</Text>
+      </View>
+    </View>
```

Comment:
it would be good to add an nlu classifier feature into here too. from my old example app:
```
      const nluModelPath = await RNFetchBlob.config({
        fileCache: true,
        appendExt: 'tflite'
      })
            .fetch('GET', 'https://d3dmqd7cy685il.cloudfront.net/nlu/production/shared/XtASJqxkO6UwefOzia-he2gnIMcBnR2UCF-VyaIy-OI/nlu.tflite')
            .then((res) => res.path())
      const nluMetadataPath = await RNFetchBlob.config({
        fileCache: true,
        appendExt: 'json'
      })
            .fetch('GET', 'https://d3dmqd7cy685il.cloudfront.net/nlu/production/shared/XtASJqxkO6UwefOzia-he2gnIMcBnR2UCF-VyaIy-OI/metadata.json')
            .then((res) => res.path())
      const nluVocabPath = await RNFetchBlob.config({
        fileCache: true,
        appendExt: 'txt'
      })
            .fetch('GET', 'https://github.com/spokestack/spokestack-ios/raw/master/SpokestackFrameworkExample/vocab.txt')
            .then((res) => res.path())

// later


  classify () {
    var input = 'How do I make pasta'
    this.onResultsChange (input)
    Spokestack.classify(input, {})
  }
```

Reply:
Nice, I can add this.

Reply:
To be added in a subsequent PR. Resolving for now.